Neos.Flow/Classes/Mvc/Dispatcher.php

```diff
@@ -70,32 +95,19 @@ public function injectObjectManager(ObjectManagerInterface $objectManager)
      * @throws NoSuchOptionException
      * @throws MissingConfigurationException
      * @api
-     * FIXME: Type hints and CLI vs Web
      */
     public function dispatch(ActionRequest $request, ActionResponse $response)
     {
-        if ($request instanceof CliRequest) {
-            $this->initiateDispatchLoop($request, $response);
-            return;
-        }
-
         // NOTE: The dispatcher is used for both Action- and CLI-Requests. For the latter case dispatching might happen during compile-time, that's why we can't inject the following dependencies
 
-        /** @var Context $securityContext */
-        $securityContext = $this->objectManager->get(Context::class);
-        if ($securityContext->areAuthorizationChecksDisabled()) {
+        if ($this->securityContext->areAuthorizationChecksDisabled()) {
```

Comment:
Ah, there are the small nice benefits of separating the two dispatchers. I think that shows this is the right direction after all!

Reply:
Indeed!